From e68897e32fb5316bdfeae41268e6efb25a5dc4ea Mon Sep 17 00:00:00 2001
From: Yerlotic <Yerlotic@users.noreply.github.com>
Date: Tue, 28 Jul 2026 13:13:09 +0000
Subject: [PATCH] Update for v0.56.1

---
 src/overview.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/overview.cpp b/src/overview.cpp
index 89aa447..13636f9 100644
--- a/src/overview.cpp
+++ b/src/overview.cpp
@@ -108,9 +108,9 @@ void HTView::hide(bool exit_on_mouse, std::optional<WORKSPACEID> target_workspac
         return;
 
     if (target_workspace.has_value()) {
-        PHLWORKSPACE workspace = g_pCompositor->getWorkspaceByID(*target_workspace);
+        PHLWORKSPACE workspace = State::workspaceState()->query().id(*target_workspace).run();
         if (workspace == nullptr && *target_workspace != WORKSPACE_INVALID)
-            workspace = g_pCompositor->createNewWorkspace(*target_workspace, monitor->m_id);
+            workspace = State::workspaceState()->create(*target_workspace, monitor->m_id);
         if (workspace == nullptr)
             return;
         monitor->changeWorkspace(workspace);
