diff --git a/cmake/Rust.cmake b/cmake/Rust.cmake
index 1de0da580..1775f9840 100644
--- a/cmake/Rust.cmake
+++ b/cmake/Rust.cmake
@@ -3,8 +3,6 @@ find_package(Rust REQUIRED)
 
 set(FISH_RUST_BUILD_DIR "${CMAKE_BINARY_DIR}/cargo/build")
 
-list(APPEND FISH_CARGO_FEATURES_LIST "embed-data")
-
 if(DEFINED ASAN)
     list(APPEND CARGO_FLAGS "-Z" "build-std")
     list(APPEND FISH_CARGO_FEATURES_LIST "asan")
