diff --git a/settings/kernel/lockdown.nix b/settings/kernel/lockdown.nix
index 71a54ca..6672104 100644
--- a/settings/kernel/lockdown.nix
+++ b/settings/kernel/lockdown.nix
@@ -27,7 +27,7 @@
       Enable linux kernel lockdown, this blocks loading of unsigned kernel modules
       and breaks hibernation.
 
-      (if false, you probably want to disable `${options.nix-mineral.settings.kernel.only-signed-modules}`)
+      (if false, you probably want to disable config.nix-mineral.settings.kernel.only-signed-modules)
     '' true;
   };
 
diff --git a/settings/kernel/only-signed-modules.nix b/settings/kernel/only-signed-modules.nix
index 40279db..e95f416 100644
--- a/settings/kernel/only-signed-modules.nix
+++ b/settings/kernel/only-signed-modules.nix
@@ -27,7 +27,7 @@
       Requires all kernel modules to be signed. This prevents out-of-tree
       kernel modules from working unless signed.
 
-      (if false, `${options.nix-mineral.settings.kernel.lockdown}` must also be false)
+      (if false, config.nix-mineral.settings.kernel.lockdown must also be false)
     '' true;
   };
 
