diff --git a/src/tools/meson.build b/src/tools/meson.build
index ffd5f15..e3a9a8d 100644
--- a/src/tools/meson.build
+++ b/src/tools/meson.build
@@ -1,7 +1,7 @@
 executable('wpctl',
   'wpctl.c',
   install: true,
-  dependencies : [gobject_dep, gio_dep, wp_dep, pipewire_dep, libintl_dep],
+  dependencies : [gobject_dep, gio_dep, wp_dep, pipewire_dep, libintl_dep, mathlib],
 )
 
 install_data('shell-completion/wpctl.bash',
diff --git a/src/tools/wpctl.c b/src/tools/wpctl.c
index 9601b89..d0cc3f0 100644
--- a/src/tools/wpctl.c
+++ b/src/tools/wpctl.c
@@ -1094,7 +1094,7 @@ set_volume_parse_positional (gint argc, gchar ** argv, GError **error)
     return FALSE;
   }
 
-  regex = g_regex_new ("^(\\d*\\.?\\d*)(%?)([-+]?)$", 0, 0, NULL);
+  regex = g_regex_new ("^(\\d*\\.?\\d*)(%|db|)([-+]?)$", 0, 0, NULL);
 
   if (g_regex_match(regex, argv[3], 0, &info)) {
     gchar *str = g_match_info_fetch (info, 1);
@@ -1103,16 +1103,30 @@ set_volume_parse_positional (gint argc, gchar ** argv, GError **error)
     g_free (str);
 
     str = g_match_info_fetch (info, 2);
-    if (g_strcmp0 (str, "%") == 0)
+    gboolean decibels = FALSE;
+    if (g_strcmp0 (str, "%") == 0) {
       cmdline.set_volume.volume /= 100;
+    } else if (g_strcmp0 (str, "db") == 0) {
+      cmdline.set_volume.volume = lround(cbrt(pow(10, cmdline.set_volume.volume / 20))*10000)/10000.0;
+      decibels = TRUE;
+    }
     g_free (str);
 
     str = g_match_info_fetch (info, 3);
-    if (g_strcmp0 (str, "-") == 0) {
-      cmdline.set_volume.volume = -(cmdline.set_volume.volume);
-      cmdline.set_volume.type = 's';
-    } else if (g_strcmp0 (str, "+") == 0) {
-      cmdline.set_volume.type = 's';
+    if (decibels) {
+      if (g_strcmp0 (str, "-") == 0) {
+        cmdline.set_volume.volume = 1/cmdline.set_volume.volume;
+        cmdline.set_volume.type = 'r';
+      } else if (g_strcmp0 (str, "+") == 0) {
+        cmdline.set_volume.type = 'r';
+      }
+    } else {
+      if (g_strcmp0 (str, "-") == 0) {
+        cmdline.set_volume.volume = -(cmdline.set_volume.volume);
+        cmdline.set_volume.type = 's';
+      } else if (g_strcmp0 (str, "+") == 0) {
+        cmdline.set_volume.type = 's';
+      }
     }
     g_free (str);
   } else {
@@ -1158,6 +1172,18 @@ do_set_volume (WpCtl * self, WpPipewireObject *proxy)
 
     new_volume += curr_volume;
   }
+  if (cmdline.set_volume.type == 'r') {
+    g_signal_emit_by_name (mixer_api, "get-volume", id, &variant);
+      if (!variant) {
+      fprintf (stderr, "Node %d does not support volume\n", id);
+      g_clear_pointer (&variant, g_variant_unref);
+      return FALSE;
+    }
+    g_variant_lookup (variant, "volume", "d", &curr_volume);
+    g_clear_pointer (&variant, g_variant_unref);
+
+    new_volume *= curr_volume;
+  }
   if (new_volume < 0) {
     new_volume = 0.0;
   }
